tests/schema/test_data_type.py

```diff
@@ -121,17 +121,34 @@ def build_sdt_fields(
             8,
             12,
         ),
+        (
+            build_sdt_fields(
+                ("ibm32", Endianness.BIG, "varA", 0),
+                ("float32", Endianness.BIG, "varB", 4),
+            ),
+            None,
+            12,
+        ),
     ],
 )
 def test_structured_data_type_descriptor(
-    fields: tuple[StructuredFieldDescriptor, ...], item_size: int, offset: int
+    fields: tuple[StructuredFieldDescriptor, ...], item_size: int | None, offset: int
 ) -> None:
     """This tests for creatin a StructuredDataTypeDescriptor for different component data types."""
     new_sdtd = StructuredDataTypeDescriptor(
         fields=list(fields), item_size=item_size, offset=offset
     )
     assert new_sdtd.dtype.names == tuple([f.name for f in fields])
-    assert new_sdtd.item_size == new_sdtd.dtype.itemsize
+    if item_size is not None:
+        assert new_sdtd.item_size == new_sdtd.dtype.itemsize
+    else:
+        assert new_sdtd.item_size == item_size
+    assert new_sdtd == StructuredDataTypeDescriptor.model_validate(
+        new_sdtd.model_dump()
+    )
+    assert new_sdtd == StructuredDataTypeDescriptor.model_validate_json(
+        new_sdtd.model_dump_json()
+    )
```

Comment:
I'm not sure if these highlighted tests are useful. We should probably hand craft the JSON strings and then check if it validates as expected.

Reply:
Okay I added a different tests for explicit conversion from a json string to StructuredDataType descriptor.

Reply:
ok cool then we can remove these because `model_validate_json` and `model_dump_json` are already tested in pydantic itself.

Reply:
Removed these lines from this test.